File: pigeons/network_discovery_printers_api.dart
import 'package:pigeon/pigeon.dart';

@HostApi()

/// {@nodoc}
abstract class NetworkDiscoveryPrintersAPI {
  List<String> subnetSearch(String subnetRange, int? waitForResponsesTimeout);

  List<String> directedBroadcast(
    String ipAddress,
    int? waitForResponsesTimeout,
  );

  List<String> localBroadcast(int? waitForResponsesTimeout);

  List<String> multicast(int hops, int? waitForResponsesTimeout);

  List<String> findPrinters(
    List<String>? printersToFind,
    int? waitForResponsesTimeout,
  );
}

File: ios/Classes/Pigeons/NetworkDiscoveryPrintersAPI.swift

class ImplNetworkDiscoveryPrintersAPI: NetworkDiscoveryPrintersAPI {
    func subnetSearch(subnetRange: String, waitForResponsesTimeout: Int64?) throws -> [String] {
        
//        do {
//            if let timeout = waitForResponsesTimeout {
//                
//                let result = try NetworkDiscoverer.subnetSearch(withRange: subnetRange, andWaitForResponsesTimeout: Int(timeout));
//                
//            }
//        } catch {
//            
//        }
        return []
    }
    
    func directedBroadcast(ipAddress: String, waitForResponsesTimeout: Int64?) throws -> [String] {
        return []
    }
    
    func localBroadcast(waitForResponsesTimeout: Int64?) throws -> [String] {
        return []
    }
    
    func multicast(hops: Int64, waitForResponsesTimeout: Int64?) throws -> [String] {
        return []
    }
    
    func findPrinters(printersToFind: [String]?, waitForResponsesTimeout: Int64?) throws -> [String] {
        return []
    }
    
    
}
